homeassistant/components/forked_daapd/media_player.py

```diff
@@ -0,0 +1,695 @@
+            intersection = [i for i in list1 + list2 if i in list1 and i in list2]
+            return bool(intersection)
+
+        _LOGGER.debug("Updating %s", update_types)
+        if "queue" in update_types:  # update queue
+            await self._update_queue()  # need to run queue before player for tts, so run first
+        # order of below don't matter
+        if intersect(["player", "options", "volume"], update_types):  # update player
+            await self._update_player()
+        if intersect(["outputs", "volume"], update_types):  # update outputs
+            await self._update_outputs()
+            self._update_zones()
+        if intersect(["database", "update", "config"], update_types):  # not supported
+            _LOGGER.debug(
+                "database/update notifications neither requested nor supported"
+            )
+        self.async_schedule_update_ha_state()
+
+    async def _update_server_config(self):
+        self._data.server_config = await self._api.get_request("config")
+
+    async def _update_player(self):
+        self._data.player = await self._api.get_request("player")
+        self._player_last_updated = utcnow()
+        self._update_track_info()
+        if self._tts_queued:
+            self._tts_playing_event.set()
+            self._tts_queued = False
+
+    async def _update_outputs(self):
+        self._data.outputs = (await self._api.get_request("outputs"))["outputs"]
+        for output in self._data.outputs:
+            if output["id"] not in self._zones:
+                self._zones[output["id"]] = ForkedDaapdZone(
+                    self, output, self._default_volume
+                )
+                self._async_add_entities([self._zones[output["id"]]], False)
+
+    def _update_zones(self):
+        for output_id, zone in self._zones.items():
+            zone.available = output_id in [
+                output["id"] for output in self._data.outputs
+            ]  # mark which zones are available
+            zone.async_schedule_update_ha_state()
+
+    async def _update_queue(self):
+        self._data.queue = await self._api.get_request("queue")
+        if (
+            self._tts_requested
+            and self._data.queue["count"] == 1
+            and self._data.queue["items"][0]["uri"].find("tts_proxy") != -1
+        ):
+            self._tts_requested = False
+            self._tts_queued = True
+        self._update_track_info()
+
+    def _update_track_info(self):  # run after every player or queue update
+        try:
+            self._data.track_info = next(
+                track
+                for track in self._data.queue["items"]
+                if track["id"] == self._data.player["item_id"]
+            )
+        except (StopIteration, TypeError, KeyError):
+            _LOGGER.debug("Could not get track info.")
+            self._data.track_info = defaultdict(str)
+
+    update_callback = partialmethod(_update)
```

Comment:
It's to allow the callback sent to the pypi library to reference self.

Reply:
Sorry, I don't understand. It does not work to give `self._update` as callback?

Reply:
You are right, I should be able to pass in self._update. I meant I used partialmethod instead of partial because I wanted to pass self. I think I was using that because I was freezing something else in the call that I took out subsequently.